keras_hub/src/models/retinanet/retinanet_object_detector.py

```diff
@@ -0,0 +1,270 @@
+        self._prediction_decoder = prediction_decoder or NonMaxSuppression(
+            from_logits=(activation != keras.activations.sigmoid),
+            bounding_box_format=bounding_box_format,
+        )
+
+        super().__init__(
+            inputs=image_input,
+            outputs=outputs,
+            **kwargs,
+        )
+
+    def compute_loss(self, x, y, y_pred, sample_weight, **kwargs):
+        y_for_label_encoder = convert_format(
+            y,
+            source=self.bounding_box_format,
+            target=self.label_encoder.bounding_box_format,
+            images=x,
+        )
+
+        boxes, classes = self.label_encoder(
+            images=x,
+            gt_boxes=y_for_label_encoder["boxes"],
+            gt_classes=y_for_label_encoder["classes"],
+        )
+
+        box_pred = y_pred["box"]
+        cls_pred = y_pred["classification"]
+
+        if boxes.shape[-1] != 4:
+            raise ValueError(
+                "boxes should have shape (None, None, 4). Got "
+                f"boxes.shape={tuple(boxes.shape)}"
+            )
+
+        if box_pred.shape[-1] != 4:
+            raise ValueError(
+                "box_pred should have shape (None, None, 4). Got "
+                f"box_pred.shape={tuple(box_pred.shape)}. Does your model's "
+                "`num_classes` parameter match your losses `num_classes` "
+                "parameter?"
+            )
+        if cls_pred.shape[-1] != self.num_classes:
+            raise ValueError(
+                "cls_pred should have shape (None, None, 4). Got "
+                f"cls_pred.shape={tuple(cls_pred.shape)}. Does your model's "
+                "`num_classes` parameter match your losses `num_classes` "
+                "parameter?"
+            )
+
+        cls_labels = ops.one_hot(
+            ops.cast(classes, "int32"), self.num_classes, dtype="float32"
+        )
+        positive_mask = ops.cast(ops.greater(classes, -1.0), dtype="float32")
+        normalizer = ops.sum(positive_mask)
+        cls_weights = ops.cast(ops.not_equal(classes, -2.0), dtype="float32")
+        cls_weights /= normalizer
+        box_weights = positive_mask / normalizer
+
+        y_true = {
+            "box": boxes,
+            "classification": cls_labels,
+        }
+        sample_weights = {
+            "box": box_weights,
+            "classification": cls_weights,
+        }
+        zero_weight = {
+            "box": ops.zeros_like(box_weights),
+            "classification": ops.zeros_like(cls_weights),
+        }
+
+        sample_weight = ops.cond(
+            normalizer == 0,
+            lambda: zero_weight,
+            lambda: sample_weights,
+        )
+        return super().compute_loss(
+            x=x, y=y_true, y_pred=y_pred, sample_weight=sample_weight, **kwargs
+        )
+
+    def predict_step(self, *args):
+        outputs = super().predict_step(*args)
+        return self.decode_predictions(outputs, args[-1])
```

Comment:
does model.fit work? can you add a test or demo to verify?

Reply:
The test case uses preprocessor and `task.predict` step as well so `predict_step` also works. If you check the test case I am expecting outputs after decoding.